botorch/utils/sampling.py

```diff
@@ -98,14 +98,12 @@ def draw_sobol_samples(
     batch_shape = batch_shape or torch.Size()
     batch_size = int(torch.prod(torch.tensor(batch_shape)))
     d = bounds.shape[-1]
-    lower = bounds[0]
-    rng = bounds[1] - bounds[0]
     sobol_engine = SobolEngine(q * d, scramble=True, seed=seed)
-    samples_raw = sobol_engine.draw(batch_size * n, dtype=lower.dtype)
-    samples_raw = samples_raw.view(*batch_shape, n, q, d).to(device=lower.device)
+    samples_raw = sobol_engine.draw(batch_size * n, dtype=bounds.dtype)
+    samples_raw = samples_raw.view(*batch_shape, n, q, d).to(device=bounds.device)
     if batch_shape != torch.Size():
         samples_raw = samples_raw.permute(-3, *range(len(batch_shape)), -2, -1)
-    return lower + rng * samples_raw
+    return bounds[0] + (bounds[1] - bounds[0]) * samples_raw
```

Comment:
same here, let's use `unnormalize` since we have it...

Reply:
The reason I reverted this having originally change to unnormalize is that unnormalize is intended as the reverse of the normalize function which has special behaviour for bounds where they're both equal. When I swapped these two cases to normalize initially I ended up breaking some multi-fidelity contribution's tests that relied on the behaviour than unnormalize with equal bounds would return the lower bound rather than 1. I could add an option to toggle that behaviour but that seems a little out of scope. https://github.com/pytorch/botorch/pull/2636#discussion_r1856802825